test/cql-pytest/test_describe.py

```diff
@@ -552,6 +552,43 @@ def test_whitespaces_in_table_options(cql, test_keyspace, scylla_only):
         desc = "\n".join([d.create_statement for d in cql.execute(f"DESC TABLE {tbl}")])
         assert re.search(regex, desc) == None
 
+# Randomly create many UDTs, with dependencies between them 
+# and validate if describe displays them in correct order.
+# UDTs should be sorted topologically, meaning if UDT `a`
+# is used to create UDT `b`, then `a` should be before `b`.
+def test_sorting(scylla_only, cql, test_keyspace, random_seed):
```

Comment:
Hmm, so you made a test which generates a single, random set of UDTs with dependencies between them... This in theory gives better coverage than a single test, but spreads it over different runs. This is prone to flakiness: a given set of UDTs may expose some bugs in the algorithm that a different set will not.

My original suggestion was to use a fixed seed and test the function on a multiple (or, a lot of) UDT sets. That would give good coverage _and_ determinism. Let's adjust the test to do that - let's choose a number of runs which does not cause the test to run for too long (not sure what is the maximum desired runtime of a cql-pytest - @nyh?).

Reply:
The seed is already fixed. `random_seed` fixture set seed for random module, prints it in logs and pass it as test's argument (so it's clearly visible in case of test failure)
I've added a loop to repeat logic of the test 10 times